### DISCUSSION TITLE
일급컬렉션의 장점, 일급 컬렉션 사용 사례

### DISCUSSION THREAD

**Discussion by 0sunset0** [cowaine/is-my-code-that-strange#68] [Chapter 07-08 (컬렉션, 강한 결합)]
책에서는 일급 컬렉션의 장점을 응집도가 높아진다고만 설명하는데 
더 많은 장점이 있을 것 같습니다. 
책에 나오지 않은 일급 컬렉션의 장점은 무엇일까요? 
그리고 일급컬렉션을 사용할 수 있는 요구사항 예시가 떠오른다면 알려주세요 🤔

**Comment by firewoody237** [comment]
- 의식적으로 디미터법칙을 지키게 됩니다. (외부에 메시지만 드러냄)
- 컬렉션에 로직이생겨 테스트가 가능해지고, 테스트코드 만드는게 쉬워집니다.
- 재활용기 가능한 컬렉션이 됩니다.

모두 응집도가 높아진 부수효과 같습니다 ㅎㅎ

**Comment by 0sunset0** [comment]
https://jojoldu.tistory.com/412
